### PR TITLE
NIFI-12282 Move Hadoop and other NARs to Optional Profiles

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-12282](https://issues.apache.org/jira/browse/NIFI-12282) Moves Apache Hadoop extensions and several other components to optional build profiles.

Changes include the following:

- Moved Hadoop and Parquet NARs to include-hadoop
- Moved HBase NARs to include-hbase
- Moved Kudu NAR to include-kudu
- Moved Solr NAR to include-solr

Apache Hadoop support varies between different vendors, which already requires custom build profiles. Both Apache Kudu and Apache Parquet components depend on several Hadoop libraries. Apache HBase and Apache Solr components each exceed 50 MB for the size of bundled NARs.

Moving these selected extensions to optional build profiles reduces the convenience binary build to around 950 MB.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-assembly/pom.xml [hunk 1]
```diff
@@ -1100,7 +1150,7 @@ language governing permissions and limitations under the License. -->
                 </dependency>
             </dependencies>
         </profile>
-	<profile>
+	    <profile>
```
Comment (pvillard31) [inline nifi-assembly/pom.xml:1153]: Expected?

File: nifi-assembly/pom.xml [hunk 2]
```diff
@@ -1115,14 +1165,14 @@ language governing permissions and limitations under the License. -->
...
+	    	    </dependency>
+		        <dependency>
                     <groupId>org.apache.nifi</groupId>
                     <artifactId>nifi-accumulo-services-api-nar</artifactId>
                     <version>2.0.0-SNAPSHOT</version>
                     <type>nar</type>
-		</dependency>
-		<dependency>
+		        </dependency>
+		        <dependency>
```
Comment (pvillard31) [inline nifi-assembly/pom.xml:1175]: Expected?

**Review by pvillard31 — COMMENTED** [review]
Can you check the below comments? Looks like unwanted indentation changes.

**Comment by exceptionfactory** [comment]
Thanks for flagging the spacing changes @pvillard31. There were a few tab characters in the nifi-assembly pom.xml, I pushed a commit to replace the rest of the tabs with spaces.

**Review by pvillard31 — APPROVED** [review]

**Pull request closed without merging**